.github/workflows/validate-rules-syntax.yaml

```diff
@@ -0,0 +1,100 @@
+on: pull_request
+permissions:
+  contents: read
+  pull-requests: read
+jobs:
+  validate_loki_recording_rules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Create tmp dir 
+        run: mkdir -p validation/recording-rules
+
+      - name: Generate Loki recording rules
+        run: yq e '.spec' loki/recording-rules.yaml > validation/recording-rules/rules.yaml
```

Comment:
is this included by default in ubuntu-latest?

Reply:
Can we wrap all rules in simple for loop in `run:` instead of creating 4 actions for different directories? Not sure how this will grow and if there will be more places where we'd add rules that need to be checked.

Reply:
> is this included by default in ubuntu-latest?


It is included in Ubuntu latest. Otherwise it would not have worked

Reply:
> Can we wrap all rules in simple for loop in `run:` instead of creating 4 actions for different directories? Not sure how this will grow and if there will be more places where we'd add rules that need to be checked, which would require having to add them to github actions steps as well.

I like the idea of having separate jobs for separate files. This allows, when a job fails, to know what is broken looking at the action name. See this screenshot for an example:

![image](https://user-images.githubusercontent.com/9251788/132205754-9e5e3f44-9463-452e-852b-9d8ef5e809ec.png)

Other issue with wrapping in into a loop, is that it will require adding more bash. I'd try to use as less bash as possible to simplify maintenance.

> if there will be more places where we'd add rules that need to be checked, which would require having to add them to github actions steps as well.

Even in the for loop case, when you add a new file, you need to add the filename to the for loop.
Unless we add even more logic for finding files with rules. That sounds like a lot of bash and complexity unneeded.


Anyway, I think you point in a valid one. Let's see what others think and reach consensus. I'm open to  modify this.

Reply:
I think I'd like the one step so that we can make this cover all rules automatically.

However I agree with making *what* failed more obvious.
How about the idea at https://github.com/grafana/cortex-rules-action#pull-request-diff where it comments on the PR?

Reply:
ok 2 against 1, we have majority.
I'm going to test how to get all rules tested in one step in an elegant way. I'll give a try to the comments in the PR too.